Opencast-Moodle/moodle-block_opencast
# Upgrade Transcription feature using event addTrack (OC 13)

This PR is meant to contain the upgrade and requirements to use the latest opencast 13 features (event addTrack).
### Prerequisites
- As for downloading the transcription files from media, you need to have LTI feature enabled [(Configure LTI)](https://docs.opencast.org/r/13.x/admin/#modules/ltimodule/#configure-lti)
- Make sure that LTI users can access 'assets/**' endpoint: 
-- In `/etc/opencast/security/mh_default_org.xml` add `ROLE OAUTH_USER` to the access of `/assets/assets/**` or provide your own more secure pattern.
![Screenshot 2023-05-22 at 17 46 25](https://github.com/Opencast-Moodle/moodle-block_opencast/assets/53179227/1ae47171-4d59-4b79-b5cc-db1cc9d92b75)


### How it works:
- It checks the opencast version and if 13 and later is detected, switches to addTrack using media to upload transcriptions.
- It still supports the old approach, which makes it easier to handle everything.
- As it is mandatory to perform LTI call to access the file, all download requests are now via LTI call.
- The setting to download via LTI is removed.
- A new setting to allow download is introduced.

### How to test this PR:
- Setting for transcriptions must be set correctly.
- Try to upload video with transcriptions.
- Make sure cron is running every minute.
- In overview page, via Manage transcription action menu item, you could see the list of existing/add new/download or delete transcriptions.
- Perform all mentioned features via the transcription management page.
- Make sure LTI feature is enabled.
